### DISCUSSION TITLE
Does using the same email across multiple verifiers result in the same key?

### DISCUSSION THREAD

**Discussion by paddyfink** [Web3Auth/Web3Auth#408] [Help]
There is this section in the docs, that is not clear :

> Should different logins lead to the same key?
> 
> You can choose to connect, logins with the same Verifier ID. For example, if a user use generic email logins and a gmail login to login on different occasions he/she can still retrieve the same key. This can only be done on logins which share a common unique identifier for a user.

1. Does that mean if I connect via email-passwordless with the email "john@doe.com" and then later, I connect via google with the same email, it will be recognized as the same user and will share the same wallet key? 

2. If I connect via google with "john@doe.com", then later I decide to switch my app to a custom authentication with auth0 for instance. if the user connects with the same email, it will be recognized as the same user and have the same key?

In both scenarios, the verifier will be different, but the identifier id (the email) will be the same.

There is also the section about `Aggregate Verifiers` in the docs. But it is not very clear and  I don't know if it is related to this use case:

> For these aggregate verifiers, its important to note that a user's Verifier ID must be known ahead of time, as keys in Web3Auth are append only. Meaning to assign to a key of a user with the aggregate verifier, Google OR Discord, we must know the user's gmail account and discord account ahead of time.

What does this mean? that the team has to merge users one by one? If I have 500 user that connects both with email-passwordless and google, do I have to contact web3auth to aggregate each user account? And I have to know those users email in advance before they log in the application?

Thanks in advance for your help.

**Comment by yashovardhan** [comment]
Hey @paddyfink 


Sorry for the delay in answering this question. It looks like your question is similar to the one answered in these two discussion threads: #549 & #421


Coming to your questions again:
> Does that mean if I connect via email-passwordless with the email "[john@doe.com](mailto:john@doe.com)" and then later, I connect via google with the same email, it will be recognized as the same user and will share the same wallet key?'

No, the keys will be different since the login methods are different, which generally means that the verifiers will be different, hence the private keys will be different.

> If I connect via google with "[john@doe.com](mailto:john@doe.com)", then later I decide to switch my app to a custom authentication with auth0 for instance. if the user connects with the same email, it will be recognized as the same user and have the same key?

Again, your login methods are different - auth0 and google. Your verifiers will differ and the private keys will differ hence.


Talking about Aggregate Verifiers:

These are basically verifiers which are a combination of other verifiers. They take some common attributes between two types of login methods, like email in your examples and club them together into a single verifier id. Currently there is no self serve way of creating a aggregate verifier. You can to [get in touch](https://web3auth.io/contact-us.html) with our team and they can help you create one.
